Mod Manager/src/routes/modList/+page.svelte

```diff
@@ -550,24 +644,83 @@
-			{#each deployDiagnostics as line}
-				<InlineNotification hideCloseButton lowContrast kind={line.includes("WARN") ? "warning" : "error"}>
-					<div slot="title" class="-mt-1 text-lg">
-						{line.includes("WARN") ? "Warning" : "Error"}
-					</div>
-					<div slot="subtitle">{line.replace(/.*WARN.*?\t/, "").replace(/.*ERROR.*?\t/, "")}</div>
-				</InlineNotification>
-			{/each}
+    <div 
+		id="deployOutputElement"
+		class="mt-2 h-[40vh] overflow-y-auto bg-neutral-800 p-2"
+        style="font-family: 'Fira Code', 'IBM Plex Mono', 'Menlo', 'DejaVu Sans Mono', 'Bitstream Vera Sans Mono', Courier, monospace; color-scheme: dark"
+		on:scroll={(event) => handleScroll(event, 'deployOutputElement')}>
+
+        {#each outputLines as line}
+            <pre class="whitespace-pre-wrap break-words text-gray-600">{@html convertAnsi.toHtml(line)}</pre>
+        {/each}
+
+        {#if userHasScrolledDeployOutput}
+            <button 
+                class="absolute top-[288px] z-10 right-[37px] bg-gray-400 hover:bg-red-500 text-white py-1 px-3 rounded"
+                on:click={enableAutoScrollDeployOutput}>
+                <svg width="24px" height="24px" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+                    <path d="M4 9l8 8 8-8" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/>
+                </svg>
+            </button>
+        {/if}
+
+    </div>
+    {#if warnings.length > 0 || errors.length > 0}
+        <br />
+        <div id="notificationElement"
+			class="flex flex-row gap-2 flex-wrap max-h-[25vh] h-[126px] overflow-y-auto relative" 
+            on:scroll={(event) => handleScroll(event, 'notificationElement')}>
+
+            {#each warnings as warning}
+                <InlineNotification hideCloseButton lowContrast kind="warning">
+                    <div slot="title" class="-mt-1 text-lg">Warning #{warning.count}</div>
+                    <div slot="subtitle">{warning.message}</div>
+                </InlineNotification>
+            {/each}
+
+            {#each errors as error}
+                <InlineNotification hideCloseButton lowContrast kind="error">
+                    <div slot="title" class="-mt-1 text-lg">Error #{error.count}</div>
+                    <div slot="subtitle">{error.message}</div>
+                </InlineNotification>
+            {/each}
+
+			{#if userHasScrolledNotifications}
+				<button 
+					class="fixed top-[479px] z-10 right-[38px] bg-gray-400 hover:bg-red-500 text-white py-1 px-3 rounded"
+					on:click={enableAutoScrollNotifications}>
+					<svg width="24px" height="24px" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
+						<path d="M4 9l8 8 8-8" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/>
+					</svg>
+				</button>
+			{/if}
+
 		</div>
 	{/if}
 
+	<div class="inline absolute right-5 text-right">
+		{#if totalNotifications  > 0}
```

Comment:
I would prefer directly using the two `length` values for this rather than 3 computed variables